### PR TITLE
Speech feedback on more Word formatting shortcuts

### LINKED ISSUE #10271: precise Announcing of the state of formatting settings when a hotkey in word is pressed
### Is your feature request related to a problem? Please describe.
No, not in the sense of a bug. It's just something I miss.
### Describe the solution you'd like
In word, for example, when I press the keystroke for bold, I would like to have NVDA say "bold on". Or "bold off", depending on what it is set to currently.
### Describe alternatives you've considered
Of course, I could just do it and check formating using NVDA+f, but it's just easier this way. Also I have a more direct confirmation.

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
Fixes (at least partially) #10271 

Supersedes #10283.
### Summary of the issue:
User wants to have font formatting state announced when native font formatting shortcut are executed under Word and Outlook.

### Description of user facing changes

The results of the following MS Word shortcuts are now reported:
* `control+shift+D`: Underline double
* `control+shift+W`: Underline words only
* `control+shift+A`: All caps
* `ctrl+Shift+K`: Small caps
* `ctrl+shift+i`, `ctrl+shift+b` and `ctrl+shift+u`: alternative gestures for toggle italic, bold and underline.

When relevant, these shortcuts are also reported in Outlook.

#### Notes about various capitalization shortcuts

* In Word, the Shift+F3 and Ctrl+Shift+A shortcuts are slightly different:
  * Shift+F3 only modifies the case of the current selection (or the word under the cursor if there is no selection)
  * Ctrl+Shift+A also modifies the case of selected text (or current word), but in addition, it modifies the case of the characters that will be typed at insertion point of the cursor.

* Outlook's Ctrl+Shift+A shortcut is equivalent to Word's Shift+F3, whereas Word's Ctrl+Shift+A has no equivalent under Outlook as far as I know.

#### Messages

Word's shortcuts Ctrl+Shift+A, Ctrl+Shift+K and Shift+F3 modify all the selection's (or current word's) capitalization. However, Ctrl+Shift+A and Ctrl+Shift+K also impact the next characters that will be typed. Thus for this shortcut "on" and "off" have been used in their announced message in order to denote a state change. On the contrary Shift+F3 announces only the result of a modification of current selection (or word); thus "on" and "off" have not been used in its message.

### Description of development approach

Use object model and polling to check state change, as done for other formatting scripts.

### Testing strategy:

Tested all combinations:
* In Word and in Outlook message (edition), English version
* For all previously listed shortcuts

### Known issues with pull request:
* This PR intially also aimed to support speech feedback for change case command, i.e. `shift+f3` (in both Word and Outlook) and `control+shift+A` in Outlook only. However, the object model property looked at for this sometimes transition through an intermediate state. Thus, this command is finally not supported by this PR, i.e. nothing is reported, rather than taking the risk to report something erroneous from time to time.
* Translators will have to modify the `gestures.ini` files in SRT repo (or elsewhere if the process changes) according to native Word or Outlook shortcuts in their language if these differ from English. I plan to inform them in detail when the beta translation freeze occurs. Or to open a separate dedicated PR if it is the new process.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

@coderabbitai summary

### CODE REVIEW

**Comment by CyrilleB79** [comment]
For the much more common underline shortcut `control+u`, we end up with "underline single" being reported instead of "underline on" before this PR. I was not very convinced to change this but Reef had recommended it at the time (https://github.com/nvaccess/nvda/pull/10283#issuecomment-878195101). What is your opinion?

**Comment by seanbudd** [comment]
@CyrilleB79 - I think to be consistent we need to solve 2 problems:
- whether to report "on" at the end in all cases. I don't think it should only happen for the single case. Right now this PR proposed to replace "on" with the style of underline. I think it would be fine to add "on" at the end after for all cases. It would be more consistent with the overall NVDA UX, but perhaps unnecessarily verbose.
- whether to be verbose reporting "single" for single underlines. I think fully dropping "single" would be fine (i.e. making the style string empty), however in that case we should probably make "on" be reported in all cases.

My preference would be:
- reporting "on" in all cases
- having an empty string for the style of single underlines

But I would accept the current alternative of:
- no longer reporting "on"
- reporting "single" for single underlines

File: source/NVDAObjects/window/winword.py [hunk 1]
```diff
@@ -1541,23 +1608,59 @@ def script_toggleItalic(self, gesture):
...
-		if val:
-			# Translators: a message when toggling formatting in Microsoft word
-			ui.message(_("Underline on"))
+		if val != WdUnderline.NONE:
+			try:
+				style = WdUnderline(val).displayString
+				if len(style) > 0:
+					style += " "
+				# Translators: a message when toggling formatting in Microsoft word
+				ui.message(_("Underline{style} on").format(style=style))
```
Comment (seanbudd) [inline source/NVDAObjects/window/winword.py:1629]: Shouldn't this be the following, the extra space is being appended?

```suggestion
				ui.message(_("Underline {style}on").format(style=style))
```

File: source/NVDAObjects/window/winword.py [hunk 2]
```diff
@@ -1541,23 +1608,59 @@ def script_toggleItalic(self, gesture):
...
+				# This may happen if the selection contains multiple underline styles and if the gesture has failed to
+				# apply the underline style (e.g. too short timeout, gesture mismatch due to localization mismatch
+				# between Word and NVDA, etc.)
+				log.debugWarning(f"No underline value for {val}")
 		else:
 			# Translators: a message when toggling formatting in Microsoft word
 			ui.message(_("Underline off"))
 
+	@script(gesture="kb:control+shift+k")
+	def script_toggleCaps(self, gesture):
```
Comment (seanbudd) [inline source/NVDAObjects/window/winword.py:1641]: please add a type hint for gesture

File: source/NVDAObjects/window/winword.py [hunk 3]
```diff
@@ -1541,23 +1608,59 @@ def script_toggleItalic(self, gesture):
 			# Translators: a message when toggling formatting in Microsoft word
 			ui.message(_("Italic off"))
 
+	@script(gestures=["kb:control+u", "kb:control+shift+u", "kb:control+shift+d"])
 	def script_toggleUnderline(self, gesture):
```
Comment (seanbudd) [inline source/NVDAObjects/window/winword.py:1612]: please add a type hint for gesture

**Review by seanbudd — COMMENTED** [review]
Generally looks good to me

**Comment by CyrilleB79** [comment]
@seanbudd thanks for your review. The changes have been applied.

**Review by seanbudd — APPROVED** [review]

**Pull request merged**